corehq/apps/ota/views.py

```diff
@@ -112,6 +112,12 @@ def app_aware_search(request, domain, app_id):
         case_search_criteria = CaseSearchCriteria(domain, case_type, criteria)
     except TooManyRelatedCasesError:
         return HttpResponse(_('Search has too many results. Please try a more specific search.'), status=400)
+    except CaseFilterError as e:
+        # This is an app building error, so show user a generic message and notify
+        notify_exception(request, str(e), details=dict(
+            exception_type=type(e),
+        ))
+        return HttpResponse(_('Could not understand search.'), status=400)
```

Comment:
Asking on the ticket for suggestions for a better error message. There isn't really anything an end user can do about this.